metron-interface/metron-rest/src/main/java/org/apache/metron/rest/config/WebSecurityConfig.java

```diff
@@ -87,13 +91,18 @@ public void configureJdbc(AuthenticationManagerBuilder auth) throws Exception {
         List<String> activeProfiles = Arrays.asList(environment.getActiveProfiles());
         if (activeProfiles.contains(MetronRestConstants.DEV_PROFILE) ||
                 activeProfiles.contains(MetronRestConstants.TEST_PROFILE)) {
-            auth.jdbcAuthentication().dataSource(dataSource)
-                    .withUser("user").password("password").roles(SECURITY_ROLE_USER).and()
-                    .withUser("user1").password("password").roles(SECURITY_ROLE_USER).and()
-                    .withUser("user2").password("password").roles(SECURITY_ROLE_USER).and()
-                    .withUser("admin").password("password").roles(SECURITY_ROLE_USER, SECURITY_ROLE_ADMIN);
+          auth.jdbcAuthentication().dataSource(dataSource)
+                  .withUser("user").password("password").roles(SECURITY_ROLE_USER).and()
+                  .withUser("user1").password("password").roles(SECURITY_ROLE_USER).and()
+                  .withUser("user2").password("password").roles(SECURITY_ROLE_USER).and()
+                  .withUser("admin").password("password").roles(SECURITY_ROLE_USER, SECURITY_ROLE_ADMIN);
         } else {
             auth.jdbcAuthentication().dataSource(dataSource);
         }
     }
+
+    @Bean
+    public PasswordEncoder passwordEncoder() {
+        return NoOpPasswordEncoder.getInstance();
```

Comment:
Admitting that I have near total ignorance here, why are we using a NoOpPasswordEncoder instead of something else? What are the implications of this?

Reply:
The reason I used that particular encoder is to keep the system working the same way it does now.  Using a different encoder would significantly alter the way we manage users and passwords.  It will also force people to migrate all passwords to a new encoding when upgrading.

I would like to see a broader discussion on our authentication strategy outside of this PR.  I suspect we will end up moving away from JDBC authentication anyways and more towards a unified architecture that aligns with other components in our stack.

Reply:
I agree we should have that discussion anyways.  I will start a thread on that topic.